rust/hyperlane-base/src/types/gcs_storage.rs

```diff
@@ -0,0 +1,245 @@
+impl fmt::Debug for GcsStorageClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("S3Storage")
+            .field("bucket", &self.bucket)
+            .finish()
+    }
+}
+
+#[async_trait]
+impl CheckpointSyncer for GcsStorageClient {
+    /// Read the highest index of this Syncer
+    async fn latest_index(&self) -> Result<Option<u32>> {
+        match self.inner.get_object(&self.bucket, LATEST_INDEX_KEY).await {
+            Ok(data) => Ok(Some(serde_json::from_slice(data.as_ref())?)),
+            Err(e) => match e {
+                // never written before to this bucket
+                ya_gcp::storage::ObjectError::InvalidName(_) => Ok(None),
+                _ => bail!(e),
+            },
+        }
+    }
+
+    /// Writes the highest index of this Syncer
+    async fn write_latest_index(&self, index: u32) -> Result<()> {
+        let d = serde_json::to_vec(&index)?;
+        self.inner
+            .insert_object(&self.bucket, LATEST_INDEX_KEY, d)
+            .await?;
+        Ok(())
+    }
+
+    /// Update the latest index of this syncer if necessary
+    async fn update_latest_index(&self, index: u32) -> Result<()> {
+        let curr = self.latest_index().await?.unwrap_or(0);
+        if index > curr {
+            self.write_latest_index(index).await?;
+        }
+        Ok(())
+    }
+
+    /// Attempt to fetch the signed (checkpoint, messageId) tuple at this index
+    async fn fetch_checkpoint(&self, index: u32) -> Result<Option<SignedCheckpointWithMessageId>> {
+        let res = self
+            .inner
+            .get_object(&self.bucket, GcsStorageClient::get_checkpoint_key(index))
+            .await?;
+        let scwmi = serde_json::from_slice(res.as_ref())?;
+        Ok(Some(scwmi))
+    }
+
+    /// Write the signed (checkpoint, messageId) tuple to this syncer
+    async fn write_checkpoint(
+        &self,
+        signed_checkpoint: &SignedCheckpointWithMessageId,
+    ) -> Result<()> {
+        self.inner
+            .insert_object(
+                &self.bucket,
+                GcsStorageClient::get_checkpoint_key(signed_checkpoint.value.index),
+                serde_json::to_vec(signed_checkpoint)?,
+            )
+            .await?;
+        Ok(())
+    }
+
+    /// Write the signed announcement to this syncer
+    async fn write_announcement(&self, signed_announcement: &SignedAnnouncement) -> Result<()> {
+        self.inner
+            .insert_object(
+                &self.bucket,
+                ANNOUNCEMENT_KEY,
+                serde_json::to_string(signed_announcement)?,
+            )
+            .await?;
+        Ok(())
+    }
+
+    /// Return the announcement storage location for this syncer
+    fn announcement_location(&self) -> String {
+        format!("gs://{}/{}", &self.bucket, ANNOUNCEMENT_KEY)
+    }
+}
+
+#[tokio::test]
+async fn public_landset_no_auth_works_test() {
+    const LANDSET_BUCKET: &str = "gcp-public-data-landsat";
```

Comment:
Can you provide some more details about what this data represents?

Reply:
it's one of the publicly available data maintained by Google used here for testing.
if there's some specific data publicly available from hyperlane maintained bucket suitable for tests - we can use it here.